test/e2e/upgrade_test.go

```diff
@@ -219,6 +226,57 @@ func TestUpgrade(t *testing.T) {
 	// test that any workloads deployed on the node have not been broken by the upgrade
 	t.Run("Workloads ready", tc.testWorkloadsAvailable)
 	t.Run("Node Logs", tc.testNodeLogs)
+	t.Run("Parallel Upgrades Checker", tc.testParallelUpgradesChecker)
+
+}
+
+// TestSetupUpgrade prep the cluster to run the upgrade tests
+func TestSetupUpgrade(t *testing.T) {
+	tc, err := NewTestContext()
+	require.NoError(t, err)
+
+	t.Run("Setup Parallel Upgrades Checker", tc.testSetupParallelUpgradesChecker)
+}
+
+// testSetupParallelUpgradesChecker tests the max number of parallel upgrades is reached. Requires 2 or more nodes
+// to run
+func (tc *testContext) testSetupParallelUpgradesChecker(t *testing.T) {
+	// get current Windows node state
+	require.NoError(t, tc.loadExistingNodes(), "error getting the current Windows nodes in the cluster")
+	nodeCount := len(gc.machineNodes) + len(gc.byohNodes)
+	if nodeCount < 2 {
+		t.Skipf("Requires 2 or more nodes to run. Found %d nodes", nodeCount)
+	}
+	ctx := context.TODO()
+	// create RBAC resources for the checker job
+	err := setupParallelUpgradesCheckerRBAC(tc.client, ctx, tc.workloadNamespace)
+	require.NoError(t, err, "error setting up RBAC for parallel upgrades checker job")
+	// deploy the checker job
+	checkerJob, err := tc.createParallelUpgradesCheckerJob()
+	require.NoError(t, err, "error creating parallel upgrades checker job")
+	log.Printf("created checker job: %s", checkerJob.GetName())
+}
+
+// testParallelUpgradesChecker tests that the number of parallel upgrades does not exceed the max allowed
+// in the lifetime of the job execution. This test is run after the upgrade is complete.
+func (tc *testContext) testParallelUpgradesChecker(t *testing.T) {
+	// get current Windows node state
+	require.NoError(t, tc.loadExistingNodes(), "error getting the current Windows nodes in the cluster")
+	nodeCount := len(gc.machineNodes) + len(gc.byohNodes)
+	if nodeCount < 2 {
+		t.Skipf("Requires 2 or more nodes to run. Found %d nodes", nodeCount)
+	}
+
+	failedPods, err := tc.client.K8s.CoreV1().Pods(tc.workloadNamespace).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: "job-name=" + parallelUpgradesCheckerJobName, FieldSelector: "status.phase=Failed"})
+	require.NoError(t, err)
+	// collect logs
+	tc.writePodLogs("job-name=" + parallelUpgradesCheckerJobName)
+	// check if any pods failed
```

Comment:
unnecessary comment, this comment is describing what the require is checking. its not adding any additional context or information.